python/nav/report/generator.py

```diff
@@ -146,25 +150,24 @@ def parse_report(self, report_name):
         the access methods will probably fit here
         """
 
-        if self.config is None:
-            self.config = io.open(self.config_file, encoding='utf-8').read()
-            self.config_local = io.open(self.config_file_local, encoding='utf-8').read()
+        if not self.config_list:
+            self.config_list = [
+                io.open(config_file, encoding="utf-8").read()
+                for config_file in self.config_files
+            ]
         report_pattern = re.compile(
             r"^\s*" + report_name + r"\s*\{(.*?)\}$", re.M | re.S | re.I
         )
-        match = report_pattern.search(self.config)
-        local_match = report_pattern.search(self.config_local)
+        matches = (report_pattern.search(config) for config in self.config_list)
+        matches = [m for m in matches if m]
 
-        if match:
+        for match in matches:
+            # Later matches will override earlier matches:
             self.parse_configuration(match.group(1))
```

Comment:
@lunkwill42 does this replace the need for the dict back-and-forth you mentioned in previous comments?

Reply:
If it is okay/cheap to re-parse reports I think it does. OK for a few reports, not OK for a lot of reports. The definition of "a lot" is very context-dependent here.

Reply:
> @lunkwill42 does this replace the need for the dict back-and-forth you mentioned in previous comments?

Not really. The code base confusingly has multiple ways of reading the config files.

The `ReportList` class reads all the config files and creates its own list of reports it found (i.e. it just scans the files for the start of every report definition and produces a list of names).  This produces a list where duplicates are intact. Secondly, it employs the `ConfigParser` class to extract titles and descriptions from these reports - which is funny, since `ConfigParser` will merge the definitions from duplicates, so the `ReportList` will end up with duplicates that have the same title and description, even if that is not how they are defined in the files.

Leaving the duplicates in there in the result from `ReportList` also kind of makes sense, if you want to trace all the files where a report is being overridden/modified.  When you are adding a file name to a `ReportTuple`, you are essentially just naming the last file that modified this report.

The `ConfigParser` class will do its own parsing of the config files, but it's mission is to return a single, cohesive report definition for any given report name.